src/business-logic/services/database-change-ingest-service.ts

```diff
@@ -40,46 +55,170 @@ export class DatabaseChangeIngestService implements IngestChangeService {
     return this.instance
   }
 
-  private readonly eventQueue: (() => Promise<void>)[] = []
+  private readonly eventPriorityQueue: Record<number, (() => Promise<void>)[]> = { }
```

Comment:
It would be nice to know if the priorities of the enqueued elements are ascending or descending.